src/contents.ts

```diff
@@ -0,0 +1,54 @@
+import { Bee } from '@ethersphere/bee-js'
+import client from 'prom-client'
+import type { ContentsConfig } from './config'
+import { logger } from './logger'
+import { register } from './metrics'
+
+const contentReuploadCounter = new client.Counter({
+  name: 'content_reupload_counter',
+  help: 'How many pinned content items were uploaded',
+})
+register.registerMetric(contentReuploadCounter)
+
+export class ContentsManager {
+  private interval?: ReturnType<typeof setInterval>
+  private isReuploading?: boolean = false
+
+  public async refreshContentReupload(beeApi: Bee): Promise<void> {
+    const pins = await beeApi.getAllPins()
+
+    if (!pins.length) {
+      logger.info(`no pins found`)
+    } else {
+      logger.info(`checking pinned content (${pins.length} pins)`)
+      pins.forEach(async pin => {
+        const isRetrievable = await beeApi.isReferenceRetrievable(pin)
+
+        if (!isRetrievable && !this.isReuploading) {
```

Comment:
Yes, this doesn't affect the procedure because if the next interval iteration starts before the current reupload finishes and there is a new pinned content unavailable it will be caught by the next iteration